SimulationAgent/SimulationRunner.cs

```diff
@@ -156,6 +161,17 @@ public async Task StartAsync(Simulation simulation)
             // of threads pending on the lock statement
             if (this.starting || this.stopping || this.running) return;
 
+            // Loop through all the device models used in the simulation
+            var models = (from model in simulation.DeviceModels where model.Count > 0 select model).ToList();
+
+            // Calculate the total number of devices
+            // the active hub, e.g. in case the user provided a custom connection string.
+            var total = models.Sum(model => model.Count);
+
+            // Send a request to update vmss auto scale settings to create vm instances
+            var vmCount = (int)Math.Ceiling((double)total / this.clusteringConfig.MaxDevicesPerNode);
```

Comment:
add a TODO here: when devices are added or removed, the number of VMs might need an update